crates/provider/src/builder.rs

```diff
@@ -99,17 +98,16 @@ impl<L, N> ProviderBuilder<L, N> {
     /// [`tower::ServiceBuilder`]. The first layer added will be the first to
     /// see the request.
     ///
-    ///
     /// [`tower::ServiceBuilder::layer`]: https://docs.rs/tower/latest/tower/struct.ServiceBuilder.html#method.layer
     /// [`tower::ServiceBuilder`]: https://docs.rs/tower/latest/tower/struct.ServiceBuilder.html
-    pub fn layer<Inner>(self, layer: Inner) -> ProviderBuilder<Stack<Inner, L>> {
+    pub fn layer<Inner>(self, layer: Inner) -> ProviderBuilder<Stack<Inner, L>, N> {
```

Comment:
need to be careful here not to ignore `N`, previously this would reset it back to `()`